includes/admin/class-wc-admin-post-types.php

```diff
@@ -961,6 +874,118 @@ public function add_display_post_states( $post_states, $post ) {
 
 		return $post_states;
 	}
+
+	/**
+	 * Apply product type constraints to stock status.
+	 *
+	 * @param WC_Product  $product The product whose stock status will be adjusted.
+	 * @param string|null $stock_status The stock status to use for adjustment, or null if no new stock status has been supplied in the request.
+	 * @return WC_Product The supplied product, or the synced product if it was a variable product.
+	 */
+	private function maybe_update_stock_status( $product, $stock_status ) {
+		if ( $product->is_type( 'external' ) ) {
+			// External products are always in stock.
+			$product->set_stock_status( 'instock' );
+		} elseif ( isset( $stock_status ) ) {
+			if ( $product->is_type( 'variable' ) && ! $product->get_manage_stock() ) {
+				// Stock status is determined by children.
+				foreach ( $product->get_children() as $child_id ) {
+					$child = wc_get_product( $child_id );
+					if ( ! $product->get_manage_stock() ) {
+						$child->set_stock_status( $stock_status );
+						$child->save();
+					}
+				}
+				$product = WC_Product_Variable::sync( $product, false );
+			} else {
+				$product->set_stock_status( $stock_status );
+			}
+		}
+
+		return $product;
+	}
+
+	/**
+	 * Set the new regular or sale price if requested.
+	 *
+	 * @param WC_Product $product The product to set the new price for.
+	 * @param string     $price_type 'regular' or 'sale'.
+	 * @return bool true if a new price has been set, false otherwise.
+	 */
+	private function set_new_price( $product, $price_type ) {
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+
+		$request_data = $this->request_data();
+
+		if ( empty( $request_data[ "change_{$price_type}_price" ] ) || ! isset( $request_data[ "_{$price_type}_price" ] ) ) {
+			return false;
+		}
+
+		$old_price     = $product->{"get_{$price_type}_price"}();
+		$price_changed = false;
+
+		$change_price  = absint( $request_data[ "change_{$price_type}_price" ] );
+		$raw_price     = wc_clean( wp_unslash( $request_data[ "_{$price_type}_price" ] ) );
+		$is_percentage = (bool) strstr( $raw_price, '%' );
+		$price         = wc_format_decimal( $raw_price );
+
+		switch ( $change_price ) {
+			case 1:
+				$new_price = $price;
+				break;
+			case 2:
+				if ( $is_percentage ) {
+					$percent   = $price / 100;
+					$new_price = $old_price + ( round( $old_price * $percent, wc_get_price_decimals() ) );
```

Comment:
[not a blocker] `round` is redundant here and on line 947 as we are doing it again on line 971 anyway.